### PR TITLE
xml filter plugin

### PR DESCRIPTION
**Pull request by GIC-de**
##### SUMMARY
The xml filter plugin allows to retrieve the text of an xml element within an xml structure provided as string. 

Example Usage: 
```yaml
  - name: show version
    junos_rpc:
        rpc: get-software-information
    register: junos_version

  - name: ...
...
     when: junos_version.xml|xml_findtext('.//package-information[name=\"junos-openconfig\"]/comment') != "JUNOS Openconfig [0.0.0.5]"
```

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
- `lib/ansible/plugins/filter/xml.py`

##### ANSIBLE VERSION
```
ansible 2.3.2.0
```

### CODE REVIEW

**Comment by dagwieers** [comment]
In the example, why is the double-quote being escaped ?

File: lib/ansible/plugins/filter/xml.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# (c) 2017, Christian Giese (@GIC-de)
```
Comment (dagwieers) [inline lib/ansible/plugins/filter/xml.py]: This should be:
```python
# Copyright: (c) 2017, Christian Giese (@GIC-de)
```

File: lib/ansible/plugins/filter/xml.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#
+# This file is part of Ansible
```
Comment (dagwieers) [inline lib/ansible/plugins/filter/xml.py]: We now have a single line GPL license statement.
```python
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```

**Comment by dagwieers** [comment]
This is looking good, and should probably get a mention in the **xml**/**win_xml** module documentation as well as [the xml module Wiki page](https://github.com/ansible/community/wiki/Module:-xml).

**Review by dagwieers — CHANGES_REQUESTED** [review]

**Review by dagwieers — APPROVED** [review]
LGTM

**Comment by GIC-de** [comment]
New `xml_findall` filter which allows to iterate over an xml structure.

```
  - name: check isis adjacency
    fail:
        msg: "isis adjacency not up"
    when: "item|xml_findtext('.//adjacency-state') != 'Up'"
    with_items: "{{ isis_adjacency.xml|xml_findall('.//isis-adjacency') }}"
```

**Comment by GIC-de** [comment]
@dagwieers is there something missing to get this change merged?

**Comment by maxamillion** [comment]
First off, apologies for the lag time on this and thank you for the pull request!

Could you please add some [integration tests](https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html) to exercise the new functionality added? Also, please include a [changelog fragment](https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#changelogs). Also, some notation in the [filters doc](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html) would be appreciated, the notable file is `./docs/docsite/rst/user_guide/playbooks_filters.rst`. Otherwise the changes look good. 👍

**Comment by maxamillion** [comment]
needs_revision

File: test/integration/targets/xml/tasks/main.yml [hunk 3]
```diff
@@ -67,3 +67,6 @@
   - include_tasks: test-get-element-content.yml
   - include_tasks: test-set-children-elements-unicode.yml
   - include_tasks: test-set-element-value-unicode.yml
+
+  # Filter tests
+  - include_tasks: test-xml-filter.yml
```
Comment (mattclay) [inline test/integration/targets/xml/tasks/main.yml:72]: Put the filter tests in the `filters` target. You'll need to update the requirements list for the test as well:

https://github.com/ansible/ansible/blob/4bd60c313bac585ca012d9729fac8e65da4b2c67/test/integration/targets/filters/runme.sh#L7

**Comment by thedoubl3j** [comment]
ping @GIC-de are you still interested in getting this merged? Looks like there is conflict with `test/integration/targets/filters/tasks/main.yml` and will need a rebase so that it can run through CI again.

**Pull request closed without merging**